src/sentry/static/sentry/app/components/emptyStateWarning.tsx

```diff
@@ -6,35 +6,33 @@ import HeroIcon from 'app/components/heroIcon';
 import InlineSvg from 'app/components/inlineSvg';
 import EmptyMessage from 'app/views/settings/components/emptyMessage';
 
-class EmptyStateWarning extends React.Component {
-  static propTypes = {
-    small: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    small: false,
-  };
-
-  render() {
-    if (this.props.small) {
-      return (
-        <EmptyMessage>
-          <SmallMessage>
-            <InlineSvg src="icon-circle-exclamation" width="34px" />
-            {this.props.children}
-          </SmallMessage>
-        </EmptyMessage>
-      );
-    }
-
-    return (
-      <EmptyStreamWrapper data-test-id="empty-state">
-        <HeroIcon src="icon-circle-exclamation" size="54" />
-        {this.props.children}
-      </EmptyStreamWrapper>
-    );
-  }
-}
+type Props = {
+  small?: boolean;
+  children: React.ReactNode;
+};
+
+const EmptyStateWarning = ({small, children}: Props) =>
```

Comment:
I would advocate for it to be standard practice for such functions be explicitly defined `React.FunctionalComponent`. 

The React component interface will change upstream with new features (e.g. deprecation of componentWillReceiveProps, `React.SFC` was deprecated to `React.FunctionalComponent` with the introduction of hooks). By reusing types from the core library and allowing those types to be enforced, we make it slightly simpler to upgrade React in the future.